src/main/scala/mesosphere/marathon/core/volume/providers/DVDI.scala

```diff
@@ -0,0 +1,163 @@
+package mesosphere.marathon.core.volume.providers
+
+import com.wix.accord._
+import com.wix.accord.dsl._
+import com.wix.accord.Validator
+import mesosphere.marathon.core.volume._
+import mesosphere.marathon.state._
+import org.apache.mesos.Protos.{ ContainerInfo, Volume => MesosVolume, Environment }
+import scala.collection.JavaConverters._
+
+/**
+  * DVDIProvider (Docker Volume Driver Interface provider) handles persistent volumes allocated
+  * by a specific docker volume driver plugin. This works for both docker and mesos containerizers,
+  * albeit with some limitations:
+  *   - only a single volume driver per container is allowed when using the docker containerizer
+  *   - docker containerizer requires that referenced volumes be created prior to application launch
+  *   - mesos containerizer only supports volumes mounted in RW mode
+  */
+protected case object DVDIProvider extends InjectionHelper[PersistentVolume] with PersistentVolumeProvider {
+  import org.apache.mesos.Protos.Volume.Mode
+
+  override val name = Some("dvdi")
+
+  val optionDriver = name.get + "/driverName"
+  val optionIOPS = name.get + "/iops"
+  val optionType = name.get + "/volumeType"
+
+  val validOptions: Validator[Map[String, String]] = validator[Map[String, String]] { opt =>
+    opt.get(optionDriver) as "driverName option" is notEmpty
+    // TODO(jdef) stronger validation for contents of driver name
+    opt.get(optionDriver).each as "driverName option" is notEmpty
+    // TODO(jdef) validate contents of iops and volume type options
+  }
+
+  val validPersistentVolume = validator[PersistentVolume] { v =>
```

Comment:
maybe validate mode RW here if we determine that docker doesn't support `:ro` suffix for driver-backed volumes